elasticsearch/class-elasticsearch.php

```diff
@@ -32,6 +32,16 @@ protected function setup_constants() {
 		if ( ! defined( 'EP_SYNC_CHUNK_LIMIT' ) ) {
 			define( 'EP_SYNC_CHUNK_LIMIT', 250 );
 		}
+
+		if ( ! defined( 'EP_HOST' ) && defined( 'VIP_ELASTICSEARCH_ENDPOINTS' ) && is_array( VIP_ELASTICSEARCH_ENDPOINTS ) ) {
+			$host = VIP_ELASTICSEARCH_ENDPOINTS[ 0 ];
```

Comment:
@darthhexx should we choose a random endpoint from the list?

Reply:
Oh, we can do better...EP provides some redundancy / retry behavior.

So we can filter `ep_pre_request_host` to walk through the list of endpoints (we can sort them randomly first for even distribution).

https://github.com/10up/ElasticPress/blob/189348ae1fd247229e984304285d472b40ee0116/includes/classes/Elasticsearch.php#L875-L887